pkg/background/generate/generator.go

```diff
@@ -161,8 +161,9 @@ func (g *generator) generate() ([]kyvernov1.ResourceSpec, error) {
 			}
 			if err != nil {
 				if !apierrors.IsAlreadyExists(err) {
-					return newGenResources, err
+					return newGenResources, fmt.Errorf("failed to create generate target resource %s/%s: %w", targetMeta.GetNamespace(), targetMeta.GetName(), err)
 				}
+				logger.V(2).Info("resource already exists, skipping creation", "resource", targetMeta.String())
```

Comment:
The if condition above only filters the alreadyExist error, we can't assume the resource already exists here. The logging message seems misleading.